unittest/client/src/ilm_common_unittests.cpp

```diff
@@ -0,0 +1,350 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <gtest/gtest.h>
+#include "ilm_common.h"
+#include "ilm_common_platform.h"
+#include "client_api_fake.h"
+
+extern "C"
+{
+    FAKE_VALUE_FUNC(ilmErrorTypes, ilmControl_init, t_ilm_nativedisplay);
+    FAKE_VOID_FUNC(ilmControl_destroy);
+    FAKE_VALUE_FUNC(ilmErrorTypes, ilmControl_registerShutdownNotification, shutdownNotificationFunc, void*);
+}
+
+class IlmCommonTest : public ::testing::Test 
+{
+public:
+    void SetUp()
+    {
+        CLIENT_API_FAKE_LIST(RESET_FAKE);
+        RESET_FAKE(ilmControl_init);
+        RESET_FAKE(ilmControl_destroy);
+        RESET_FAKE(ilmControl_registerShutdownNotification);
+    }
+
+    void TearDown()
+    {
+        if(ilm_isInitialized())
+        {
+            ilm_destroy();
+        }
+    }
+
+    void mock_ilmInitSuccess()
+    {
+        mpp_wlDisplays[0] = (struct wl_display*)&m_wlDisplayFakePointer;
+        SET_RETURN_SEQ(wl_display_connect, mpp_wlDisplays, MAX_NUMBER);
+
+        mp_ilmErrorType[0] = ILM_SUCCESS;
+        SET_RETURN_SEQ(ilmControl_init, mp_ilmErrorType, MAX_NUMBER);
+    }
+
+    static constexpr uint8_t MAX_NUMBER = 1;
+    uint8_t m_wlDisplayFakePointer = 0;
+    struct wl_display* mpp_wlDisplays [MAX_NUMBER] = {nullptr};
+    ilmErrorTypes mp_ilmErrorType[MAX_NUMBER] = {ILM_FAILED};
+};
+
+TEST_F(IlmCommonTest, ilm_init_cannotGetDisplay)
+{
+    // Prepare fake for wl_display_connect, to make the result is nullptr
+    mpp_wlDisplays[0] = nullptr;
+    SET_RETURN_SEQ(wl_display_connect, mpp_wlDisplays, MAX_NUMBER);
+    // Invoke the ilm_init and expect the result is ILM_FAILED
+    ASSERT_EQ(ILM_FAILED, ilm_init());
+    /* Check the logic
+     * wl_display_connect should call one time and return nullptr 
+     */
+    ASSERT_EQ(wl_display_connect_fake.call_count, 1);
+    ASSERT_EQ(wl_display_connect_fake.return_val_history[0], nullptr);
+}
+
+TEST_F(IlmCommonTest, ilm_init_getFailedOnilmControl_init)
+{
+    // Prepare fake for wl_display_connect, to make return different nullptr
+    mpp_wlDisplays[0] = (struct wl_display*)&m_wlDisplayFakePointer;
+    SET_RETURN_SEQ(wl_display_connect, mpp_wlDisplays, MAX_NUMBER);
+    // Prepare fake for ilmControl_init, to make return ILM_FAILED
+    mp_ilmErrorType[0] = ILM_FAILED;
+    SET_RETURN_SEQ(ilmControl_init, mp_ilmErrorType, MAX_NUMBER);
+    // Invoke the ilmControl_init and expect the result is ILM_FAILED
+    ASSERT_EQ(ILM_FAILED, ilm_init());
+    /* Check the logic
+     * ilmControl_init_fake should call one time and return ILM_FAILED
+     */
+    ASSERT_EQ(ilmControl_init_fake.call_count, 1);
+    ASSERT_EQ(mp_ilmErrorType[0], ilmControl_init_fake.return_val_history[0]);
+}
+
```

Comment:
It would be beneficial to have a concise description and goal of each test case.

Reply:
Hi @HarshaMM, Thanks for your comment.
I added a concise description and goal to some test cases and remove comments inside tests. Please review it for me.
And I will change it to all of the tests if you approve.